src/beamlime/dashboard/widgets/job_status_widget.py

```diff
@@ -0,0 +1,547 @@
+            button_type="light",
+            margin=bnt_margin,
+        )
+        self._bulk_reset_btn.on_click(self._bulk_reset)
+
+        self._bulk_stop_btn = pn.widgets.Button(
+            name=f"{UIConstants.STOP_SYMBOL} Stop Selected",
+            button_type="light",
+            margin=bnt_margin,
+        )
+        self._bulk_stop_btn.on_click(self._bulk_stop)
+
+        # Controls row
+        self._controls = pn.Row(
+            self._select_all_btn,
+            self._select_none_btn,
+            self._invert_selection_btn,
+            pn.Spacer(width=20),
+            self._bulk_reset_btn,
+            self._bulk_stop_btn,
+            margin=(5, 10),
+        )
+
+        self._job_list = pn.Column(sizing_mode="stretch_width", margin=(0, 10))
+
+        # Initialize with current job statuses
+        for job_status in self._job_service.job_statuses.values():
+            self._add_or_update_job_widget(job_status)
+
+    def _on_status_update(self) -> None:
+        """Handle job status updates from the service."""
+        # Get all current job statuses and update widgets accordingly
+        current_statuses = self._job_service.job_statuses
+        current_job_keys = {
+            f"{status.job_id.source_name}:{status.job_id.job_number}"
+            for status in current_statuses.values()
+        }
+
+        # Remove widgets for jobs that no longer exist
+        widgets_to_remove = set(self._status_widgets.keys()) - current_job_keys
+        for job_key in widgets_to_remove:
+            self._remove_job_widget(job_key)
+
+        # Add or update widgets for current jobs
+        for job_status in current_statuses.values():
+            self._add_or_update_job_widget(job_status)
+
+    def _add_or_update_job_widget(self, job_status: JobStatus) -> None:
+        """Add a new job widget or update an existing one."""
+        job_key = f"{job_status.job_id.source_name}:{job_status.job_id.job_number}"
+
+        if job_key in self._status_widgets:
+            # Update existing widget
+            self._status_widgets[job_key].update_status(job_status)
+        else:
+            # Create new widget
+            widget = JobStatusWidget(job_status, self._job_controller)
+            widget_panel = widget.panel()
+            self._status_widgets[job_key] = widget
+            self._widget_panels[job_key] = widget_panel
+            self._job_list.append(widget_panel)
+
+    def _remove_job_widget(self, job_key: str) -> None:
+        """Remove a job widget."""
+        if job_key in self._status_widgets:
+            self._status_widgets.pop(job_key)
+            if job_key in self._widget_panels:
+                widget_panel = self._widget_panels.pop(job_key)
+                # Remove from the job list
+                try:
+                    self._job_list.remove(widget_panel)
+                except ValueError:
+                    # Panel might not be in the list anymore
```

Comment:
What happens if Panel is not in the list anymore? Can you describe in which situation this arises?

Reply:
Adding a comment.